src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1985,6 +1988,17 @@ public void DeleteWithoutCreatingTombstone(DocumentsOperationContext context, st
                         ref oldValue, newEtag);
                 }
 
+                if (collectionName.IsSystem == false && (flags & DocumentFlags.Artificial) != DocumentFlags.Artificial)
+                {
+                    bool hasVersion =
+                        _documentDatabase.BundleLoader.VersioningStorage?.PutFromDocument(context, collectionName,
+                        key, document,changeVector) ?? false;
```

Comment:
If we remove this, we will send on replication the same document with different change vector.
Which will result on the destination as a document with 2 different versions.